source/docs/guide/src/ide_support.md

```diff
@@ -0,0 +1,146 @@
+
+The propagated verification error will generate red squiggles at the corresponding code location. In VS Code, the `View > Problems` tab will list the entire set of diagnostics generated by Verus. Error messages that are not from diagnostic will be appended to the Verus result notification.
+
+#### 4.Verus code actions
+Code actions try to automate common code rewrites. When a user puts the cursor on a certain location, rust-analyzer calculates the list of applicable code actions. After the user clicks the lightbulb icon, the user can apply one of the actions in the list. If the lightbulb does not show up, it means that there are no applicable code actions at that code location. 
+
+If the action includes invoking Verus, it is may take a few seconds after you apply the code action.
+
+
+##### 4.1. `Add proof block for this assert`
+This action is applicable when the cursor is on `assert`.
+It introduces `by` keyword and proof block. It also copies the assertion inside the new proof block. This rewrite action is not applicable when the assertion already has a `by` keyword.
+
+
+##### 4.2. `Check if this assertion is essential`
+This action is applicable when the cursor is on `assert`.
+
+It checks if this function still verifies without that assertion. It tries to verify the surrounding function after removing that assertion. To be specific, it runs with `--verify-function` flag. If it fails, it will add `// OBSERVE` comment to indicate that this assertion matters. If the function still verifies, the assertion will be commented out. 
+
+This code action assumes that the surrounding function is already verified. Hence when the surrounding function is not verified, it will mark that assertion is important even if it is purely redundant. Ideally, this action could first verify the surrounding function and then try out without that assertion, but it will increase response time significantly.
+
+
+##### 4.3. `Copy ensures clauses to the end`
+This action is applicable when the cursor is on `ensures`.
+It copies the ensures clauses to the end of the function. If the function has a return value, that value will be let-binded using the identifier of the return value.  
+
+
+##### 4.4. `Insert requires clauses of this function call`
+This action is applicable when the cursor is on a function call that has requires clauses.
+It inserts the requires clauses and inlines the arguments.
+
+
+##### 4.5. `Inline function`
+This action is applicable when the cursor is on a function call.
+This is already implemented in the original rust-analyzer. Although this is not Verus-specific, it is worth noting that this could be handy for debugging proofs.
+
+
+
+
+
+
+
+## Limitations 
+General limitations are the following:
+1. This is currently experimental, and only tested using VS Code on macOS.  
```

Comment:
Can you add this note at the top of the file? I'd rather have people know this is more on the experimental side for the time being, rather then have them go through the guide expecting a smooth experience on all platforms.